### ISSUE TITLE
File-Uploader

### ISSUE THREAD

**Issue by KlonAnon** [telekom/scale#1645]
**Description and use case - File-Uploader**
I'd like to suggest a component to upload files. Maybe this already exists and I overlooked it and in that case I'd be sorry.
In my opinion this is a component that is fairly often used. Files should be imported via a dropzone (drag and drop) or by choosing it form the computer filesystem. There should be the possibility to trigger a function when inserting the file or after pressing a button (the file could be uploaded directly after dropping it or after pressing a button, I don't know which one is better). The function should be something the developer defines himself (e.g. the logic to upload it to the server/database). Also I think there should be a way make sure only the right files can be inserted (maybe it should only be allowed to upload a certain filetype or filesize)

**Alernatives I've considered**
While it is possible to implement to implement this on your own for example with a scale-modal as basis , I think it would be somethink nice to have directly. As I already mentioned, I think this is a component that needs to be implemented fairly often, so I think it would save time for the Telekom-Developer-Community to have this component implemented within  this library.

**Additional context**
This component could possibly look something like this:
![image](https://user-images.githubusercontent.com/59443262/224106682-76f6e8bd-5993-4f02-b8c3-9d40527652c1.png)

**Comment by Vettl5** [comment]
True

**Comment by Traciges** [comment]
Wichtig und richtig

**Comment by robertgebhard** [comment]
Yes please we are building something similar. A one-size-fits-all solution would be awesome.
<img width="1222" alt="Bildschirmfoto 2023-09-06 um 14 23 33" src="https://github.com/telekom/scale/assets/95682620/6468de1b-ae33-48b8-95d6-0c1743f385fc">

**Comment by Darkavid** [comment]
any update on this one?
